app/forms/providers/means/capital_disregards/mandatory_form.rb

```diff
@@ -0,0 +1,115 @@
+
+        validate :validate_any_checkbox_checked, unless: :draft?
+        validate :validate_none_selected_and_another_checkbox_not_both_checked, unless: :draft?
+
+        def initialize(params = {})
+          self.legal_aid_application = params.delete(:model)
+          self.mandatory_capital_disregards = params["mandatory_capital_disregards"] || existing_mandatory_disregards
+
+          super
+        end
+
+        def mandatory_capital_disregards=(names)
+          @mandatory_capital_disregards = [names].flatten.compact_blank
+        end
+
+        def save_as_draft
+          @draft = true
+          save!
+        end
+
+        def draft?
+          @draft
+        end
+
+        def save
+          return false unless valid?
+
+          destroy_previous_disregards!
+          create_new_disregards!
+        end
+        alias_method :save!, :save
+
+      private
+
+        # create new records if they don't already exist
+        def create_new_disregards!
+          return unless mandatory_capital_disregards
+
+          mandatory_capital_disregards&.reject { |disregard|
+            existing_mandatory_disregards.include?(disregard)
+          }&.each { |disregard| legal_aid_application.capital_disregards.create!(name: disregard, mandatory: true) }
+        end
+
+        # destroy only the records which already exist but aren't selected
+        def destroy_previous_disregards!
+          legal_aid_application.mandatory_capital_disregards&.where&.not(name: mandatory_capital_disregards)&.destroy_all
+        end
+
+        def existing_mandatory_disregards
+          legal_aid_application.mandatory_capital_disregards.pluck(:name)
+        end
+
+        def none_selected?
+          none_selected == "true"
+        end
+
+        def any_checkbox_checked?
+          none_selected? || mandatory_capital_disregards.any?
+        end
+
+        def none_and_another_checkbox_checked?
+          none_selected? && mandatory_capital_disregards.any?
+        end
+
+        def validate_any_checkbox_checked
+          errors.add :mandatory_capital_disregards, error_message_for_none_selected unless any_checkbox_checked?
+        end
+
+        def validate_none_selected_and_another_checkbox_not_both_checked
+          errors.add :mandatory_capital_disregards, error_message_for_none_and_another_option_selected if none_and_another_checkbox_checked?
+        end
+
+        def error_message_for_none_selected
+          I18n.t("activemodel.errors.models.mandatory_capital_disregards.attributes.base.none_selected")
+        end
+
+        def error_message_for_none_and_another_option_selected
+          I18n.t("activemodel.errors.models.mandatory_capital_disregards.attributes.base.#{error_key('none_and_another_option_selected')}")
+        end
+
+        def error_key(key_name)
+          return "#{key_name}_with_partner" if legal_aid_application&.applicant&.has_partner_with_no_contrary_interest?
+
+          key_name
+        end
```

Comment:
Does this form inherit `app/forms/base_form.rb`

If yes, could/should we use the `error_key` method from the base form?

Reply:
No it doesn't inherit from base_form, there was a conversation yesterday about the base form having too much 'rails magic' causing issues!

Reply:
It does not inherit BaseForm, it follows the same pattern as the (now merged) `discretionary_disregards` page and the existing transactions pages , which are a bit different to the model-based forms